common/can/can.h

```diff
@@ -0,0 +1,58 @@
+#ifndef CAN_H_
+#define CAN_H_
+
+#include "ch.h"
+#include "hal.h"
+#include "canard.h"
+
+// ---- Implementation Requirements ---- //
+struct can_msg_handler {
+  uint16_t id;
+  uint64_t signature;
+  void (*handler)(CanardInstance* ins, CanardRxTransfer* transfer);
+};
+
+#define CAN_MSG_HANDLER(topic_id, topic_signature, topic_handler) \
+  { .id = topic_id, .signature = topic_signature, .handler = topic_handler }
+#define CAN_MSG_HANDLER_END \
+  { .id = 0, .signature = 0, .handler = NULL }
+
+extern struct can_msg_handler can_broadcast_handlers[];
```

Comment:
Is there a better alternative we can do for this? This dependency is going to be very unclear when using this module.

Maybe store a pointer and have a function to set it? Or a function to subscribe a handler?

Maybe as part of `can_init`?

Reply:
Yeah, I think `can_init` is best here, it shouldn't change during runtime and if we're going to move it into a function I want to capture it as an explicit dependency (i.e. you can easily forget to call another function to set this - and as it stands you will get a linker error if you forget to attach these lists).

I wasn't thinking too hard about the API because libcanard v1 sidesteps this with a specific function to register subscriptions.

Reply:
Resolving this under the assumption you took a look when accepting the PR.